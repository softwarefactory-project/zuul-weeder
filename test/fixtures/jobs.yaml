- job:
    name: base
    parent: null
    description: The base job.
    pre-run: playbooks/base/pre.yaml
    post-run:
      - playbooks/base/post.yaml
    roles:
      - zuul: sf-jobs
      - zuul: zuul-jobs
    extra-vars:
      zuul_use_fetch_output: true
    timeout: 1800
    attempts: 3
    secrets:
      - site_sflogs
    nodeset:
      nodes:
        - name: container
          label: pod-centos-7

- job:
    name: config-check
    parent: base
    final: true
    dependencies:
      - job1
      - name: job2
        soft: true
    allowed-projects:
      - config
    description: Validate the config repo.
    run: playbooks/config/check.yaml
    post-run: playbooks/config/check-fetch-artifacts.yaml
    branches: "master"
    secrets:
      - service_user
    vars:
      zuul_log_url: "https://sftests.com/logs"
      gateway_url: "https://sftests.com"
      tenant_config: False
    nodeset:
      nodes:
        name: test-node
        label: cloud-fedora-35

- job:
    name: config-update
    parent: base
    final: true
    allowed-projects:
      - config
    description: Deploy config repo update.
    run: playbooks/config/update.yaml
    branches:
      - "master"
    secrets:
      - site_install_server
    semaphore: semaphore-config-update
    nodeset:
      nodes: []

- job:
    name: wait-for-changes-ahead
    parent: null
    timeout: 7200
    nodeset: nodeset1
    vars:
      zuul_web_url: "https://sftests.com/zuul"
    description: |
      This job wait for the queue ahead to be empty.

      Responds to these variables:

      .. zuul:jobvar:: zuul_web_url

         The zuul web api url.

      .. zuul:jobvar:: wait_timeout
         :default: 120

         Wait timeout in minutes.

    run: playbooks/wait-for-changes-ahead.yaml