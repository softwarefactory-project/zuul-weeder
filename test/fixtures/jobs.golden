[ ZJob
    ( Job
        { name = JobName "base"
        , parent = Nothing
        , nodeset = Just
            ( JobAnonymousNodeset
                [ NodeLabelName "pod-centos-7" ]
            )
        , branches = []
        , dependencies = []
        }
    )
, ZJob
    ( Job
        { name = JobName "config-check"
        , parent = Just
            ( JobName "base" )
        , nodeset = Just
            ( JobAnonymousNodeset
                [ NodeLabelName "cloud-fedora-35" ]
            )
        , branches =
            [ BranchName "master" ]
        , dependencies =
            [ JobName "job1"
            , JobName "job2"
            ]
        }
    )
, ZJob
    ( Job
        { name = JobName "config-update"
        , parent = Just
            ( JobName "base" )
        , nodeset = Just
            ( JobAnonymousNodeset [] )
        , branches =
            [ BranchName "master" ]
        , dependencies = []
        }
    )
, ZJob
    ( Job
        { name = JobName "wait-for-changes-ahead"
        , parent = Nothing
        , nodeset = Just
            ( JobNodeset
                ( NodesetName "nodeset1" )
            )
        , branches = []
        , dependencies = []
        }
    )
]